File: Documentation/API/Parsers/Chart/ReadTempoChangesFromChartData.cpp
#include <iostream>

#include "RhythmGameUtilities/File.hpp"
#include "RhythmGameUtilities/Parsers/Chart.hpp"

using namespace RhythmGameUtilities;

auto main() -> int
{
    auto contents = ReadStringFromFile("./song.chart");

    auto tempoChanges = ReadTempoChangesFromChartData(contents.c_str());

    std::cout << size(tempoChanges) << std::endl; // 7

    return 0;
}

File: Documentation/API/Parsers/Chart/ReadTempoChangesFromChartData.js
import { readFileSync } from 'node:fs';

import RhythmGameUtilities from '@rhythm-game-utilities/core';

const contents = readFileSync('./tests/mocks/song.chart', 'utf-8');

const tempoChanges = RhythmGameUtilities.ReadTempoChangesFromChartData(contents);

console.log(tempoChanges.length); // 7
